src/renderer/widgets/Transfer/default/model/form-model.ts

```diff
@@ -425,36 +428,28 @@ sample({
   target: $errors,
 });
 
-const $totalFee = combine(
+// available balance
+
+const $availableBalance = combine(
   {
-    validationDone: $validationDone,
-    validationResults: $balanceValidationResults,
-    asset: $asset,
-    initiator: form.fields.initiator.$value,
+    availableBalances: $availableBalances,
+    balance: $initiatorAccountBalance,
   },
-  ({ validationResults, asset, initiator, validationDone }) => {
-    if (!validationDone || nullable(asset) || nullable(initiator)) {
-      return null;
-    }
-
-    return combineTotalRequiredFee({
-      validationResults,
-      account: initiator,
-      assetId: asset.assetId,
-      excludeActions: ['sending amount'],
-    });
+  ({ availableBalances, balance }) => {
+    if (nullable(balance)) return null;
+    return availableBalances.find((b) => b.id === balance.id) ?? balance;
   },
 );
 
 sample({
   source: {
-    balance: $initiatorAccountBalance,
-    totalFee: $totalFee,
-    isExistentialDepositEnabled: $isExistentialDepositEnabled,
+    balance: $availableBalance,
+    balancePreservationStrategy: $balancePreservationStrategy,
   },
-  filter: ({ balance, totalFee }) => nonNullable(balance) && nonNullable(totalFee),
-  fn: ({ balance, totalFee, isExistentialDepositEnabled }) => {
-    return getAvailableAmount({ balance: balance!, totalFee: totalFee!, includeED: isExistentialDepositEnabled });
+  fn: ({ balance, balancePreservationStrategy }) => {
+    if (nullable(balance)) return BN_ZERO;
```

Comment:
@johnthecat sometimes we face issues because of syntetic (and incorrect) zero fallbacks

maybe filtering was a good approach?